### PR TITLE
name D_800DF4E0->gNeedToCloseDialogueBox

### PR DESCRIPTION
**Pull request by tonyspumoni**
Identified w/ project64 debugger/breakpoints

### CODE REVIEW

**Review by DavidSM64 — APPROVED** [review]
LGTM

File: src/menu.c [hunk 1]
```diff
@@ -328,7 +328,7 @@ s32 D_800DF4D0              = 0; // Unused?
 s32 gTrackIdToLoad          = 0;
 s8 D_800DF4D8               = 1;
 s8 D_800DF4DC               = 0;
-s8 D_800DF4E0               = 0;
+s8 gNeedToCloseDialogueBox  = 0;
```
Comment (FazanaJ) [inline src/menu.c:331]:
```suggestion
s8 gNeedToCloseDialogueBox  = FALSE;
```

File: src/menu.c [hunk 2]
```diff
@@ -7800,14 +7800,14 @@ void func_8009CF68(s32 arg0) {
             sCurrentMenuID = 0;
             sDialogueOption = 0;
         }
-        D_800DF4E0 = 1;
+        gNeedToCloseDialogueBox = 1;
```
Comment (FazanaJ) [inline src/menu.c:7803]:
```suggestion
        gNeedToCloseDialogueBox = TRUE;
```

File: src/menu.c [hunk 3]
```diff
@@ -7800,14 +7800,14 @@ void func_8009CF68(s32 arg0) {
...
         D_800DF4E4[arg0] = 1;
     }
 }
 
 void func_8009CFB0(void) {
-    if (D_800DF4E0 != 0) {
-        D_800DF4E0 = 0;
+    if (gNeedToCloseDialogueBox != 0) {
```
Comment (FazanaJ) [inline src/menu.c:7809]:
```suggestion
    if (gNeedToCloseDialogueBox) {
```

File: src/menu.c [hunk 4]
```diff
@@ -7800,14 +7800,14 @@ void func_8009CF68(s32 arg0) {
...
+        gNeedToCloseDialogueBox = 0;
```
Comment (FazanaJ) [inline src/menu.c:7810]:
```suggestion
        gNeedToCloseDialogueBox = FALSE;
```

File: src/menu.c [hunk 5]
```diff
@@ -7820,7 +7820,7 @@ s32 func_8009CFEC(u32 dialogueOption) {
     if ((func_800C3400() != 0) && (dialogueOption != DIALOG_CHALLENGE)) {
         return 0;
     }
-    if (D_800DF4E0 != 0) {
+    if (gNeedToCloseDialogueBox != 0) {
```
Comment (FazanaJ) [inline src/menu.c:7823]:
```suggestion
    if (gNeedToCloseDialogueBox) {
```

File: src/menu.c [hunk 6]
```diff
@@ -8082,7 +8082,7 @@ s32 taj_menu_loop(void) {
             if (sCurrentMenuID == 7) {
                 sp2C = 3;
             }
-            D_800DF4E0 = 0;
+            gNeedToCloseDialogueBox = 0;
```
Comment (FazanaJ) [inline src/menu.c:8085]:
```suggestion
            gNeedToCloseDialogueBox = FALSE;
```

**Review by FazanaJ — COMMENTED** [review]
Otherwise all good

**Pull request merged**